cmd/gocached/gocached.go

```diff
@@ -579,32 +746,119 @@ func (s *server) handlePut(w http.ResponseWriter, r *http.Request) {
 	)
 	if err != nil {
 		s.logf("Actions insert error: %v", err)
-		s.PutErrs.Add(1)
+		stats.putErrs++
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	affected, err := res.RowsAffected()
 	if err != nil {
 		s.logf("Actions rows affected error: %v", err)
-		s.PutErrs.Add(1)
+		stats.putErrs++
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	if affected == 0 {
-		s.PutsDup.Add(1)
+		stats.putsDup++
 	}
 
-	s.Puts.Add(1)
-	s.PutsBytes.Add(r.ContentLength)
+	stats.puts++
+	stats.putsBytes += r.ContentLength
 	if smallData != nil {
-		s.PutsInline.Add(1)
+		stats.putsInline++
 	}
 
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// handleTokenExchange handles POST /auth/exchange-token requests to exchange
+// a JWT for an access token. Each access token represents a session that will
+// last for one hour and have cache stats associated with it.
+func (srv *server) handleTokenExchange(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
```

Comment:
http.MaxBytesReader to bound this read size?

Reply:
Based on a little bit of research I picked 4KiB as a reasonable upper bound, but we can always bump it if needed